lib/bridge/MatrixHandler.js

```diff
@@ -853,9 +856,36 @@ MatrixHandler.prototype._onJoin = Promise.coroutine(function*(req, event, user)
         }
         // get the virtual IRC user for this user
         promises.push(Promise.coroutine(function*() {
-            let bridgedClient = yield self.ircBridge.getBridgedClient(
-                room.server, user.getId(), (event.content || {}).displayname
-            );
+            let bridgedClient;
+            let kickIntent;
+            try {
+                bridgedClient = yield self.ircBridge.getBridgedClient(
+                    room.server, user.getId(), (event.content || {}).displayname
+                );
+            }
+            catch (e) {
+                // We need to kick on failure to get a client.
+                req.log.info(`${user.getId()} failed to get a IRC connection. Kicking from room.`);
+                kickIntent = self.ircBridge.getAppServiceBridge().getIntent();
+
+            }
+
+            while (kickIntent) {
+                try {
+                    yield kickIntent.kick(
+                        event.room_id, user.getId(),
+                    `Connection limit reached for ${room.server.domain}. Please try again later.`
```

Comment:
can we indent this (which pushes it on another line, which is annoying)?

Reply:
We can't, because we go over the silly 100 char limit :(